torch/testing/__init__.py

```diff
@@ -106,8 +114,19 @@ def _compare_tensors_internal(a: torch.Tensor, b: torch.Tensor, *, rtol, atol, e
     # Compares complex tensors' real and imaginary parts separately.
     # (see NOTE Test Framework Tensor "Equality")
     if a.is_complex():
+        a = a.clone()
```

Comment:
This clone's only needed if equal_nan is "relaxed", right? 

Also this `a_real = a.real` nonsense is just an artifact of a more complicated workaround that's no longer necessary. That was fixed in this PR https://github.com/pytorch/pytorch/pull/39065/files, so would you please get rid of a_real and a_imag and b_real and b_imag? We should just a.real and a.imag and etc.

Reply:
With those changes I think this simplifies to:

```
if equal_nan == "relaxed":
  a = a.clone()
  b = b.clone()
  a.real[a.imag.isnan()] = math.nan
  a.imag[a.real.isnan()] = math.nan
  b.real[b.imag.isnan()] = math.nan
  b.imag[b.real.isnan()] = math.nan
```